pkg/trait/knative.go

```diff
@@ -214,6 +228,12 @@ func (t *knativeTrait) getConfiguredSourceChannels() []string {
 }
 
 func (*knativeTrait) getSourceChannels(e *environment) []string {
-	meta := metadata.Extract(e.Integration.Spec.Source)
-	return knativeutil.ExtractChannelNames(meta.FromURIs)
+	channels := make([]string, 0)
+
+	for _, s := range e.Integration.Spec.Sources {
+		meta := metadata.Extract(s)
+		channels = append(channels, knativeutil.ExtractChannelNames(meta.FromURIs)...)
```

Comment:
Maybe we should consider adding to the `meta` package some functions to inpect all sources without iterating on them each time.. For the future..